sigstore/sign.py

```diff
@@ -188,3 +212,61 @@ class SigningResult(BaseModel):
     """
     A record of the Rekor log entry for the signing operation.
     """
+
+    def _to_bundle(self) -> Bundle:
+        """
+        Creates a Sigstore bundle (as defined by Sigstore's protobuf specs)
+        from this `SigningResult`.
+        """
+
+        # TODO: Should we include our Fulcio intermediate in the chain?
+        cert = x509.load_pem_x509_certificate(self.cert_pem.encode())
+        cert_der = cert.public_bytes(encoding=serialization.Encoding.DER)
+        chain = X509CertificateChain(certificates=[X509Certificate(raw_bytes=cert_der)])
+
+        inclusion_proof: InclusionProof | None = None
+        if self.log_entry.inclusion_proof is not None:
+            inclusion_proof = InclusionProof(
+                log_index=self.log_entry.inclusion_proof.log_index,
+                root_hash=bytes.fromhex(self.log_entry.inclusion_proof.root_hash),
+                tree_size=self.log_entry.inclusion_proof.tree_size,
+                hashes=[
+                    bytes.fromhex(h) for h in self.log_entry.inclusion_proof.hashes
+                ],
+            )
+
+        tlog_entry = TransparencyLogEntry(
+            log_index=self.log_entry.log_index,
+            log_id=LogId(key_id=bytes.fromhex(self.log_entry.log_id)),
+            # TODO: Maybe leave this field out? It appears to be optional
+            # and it's just hardcoded for us, since it's the only kind of Rekor
+            # entry we support.
+            kind_version=KindVersion(kind="hashedrekord", version="0.0.1"),
+            integrated_time=self.log_entry.integrated_time,
+            inclusion_promise=InclusionPromise(
+                signed_entry_timestamp=base64.b64decode(
+                    self.log_entry.signed_entry_timestamp
+                )
+            ),
+            inclusion_proof=inclusion_proof,
+            canonicalized_body=base64.b64decode(self.log_entry.body),
+        )
+
+        material = VerificationMaterial(
+            x509_certificate_chain=chain,
+            tlog_entries=[tlog_entry],
+        )
+
+        bundle = Bundle(
+            media_type="application/vnd.dev.sigstore.bundle+json;version=0.1",
```

Comment:
It'd be nice if there was some way to default to this `media_type`, but I have no idea if that's possible :slightly_smiling_face:

Reply:
IIUC no :(

That said, I think that's sort-of by design? You should only set that `media_type` if you immediately plan to serialize to JSON

Reply:
Aha, I missed that detail! That makes sense, then, and it's not a huge deal here.

Reply:
Yeah, it would be nice to have const values in the language bindings that would capture the current media type used.

Reply:
100% agreed -- I'm not sure how best to accomplish that without hacky patches to the codegen, though...

Reply:
Yes, would be messy. I would love for Protobufs to support that natively, so many use-cases I have seen where it would make our lives easier.